src/DataFixtures/MajorVersionFixtures.php

```diff
@@ -68,18 +87,21 @@ public static function getVersions(): array
         ];
     }
 
+    /**
+     * @return array<string, array<string, float|int|\DateTimeImmutable|\Doctrine\Common\Collections\ArrayCollection<mixed, mixed>|string|null>>
+     */
```

Comment:
This return statement perfectly demonstrates why it should be returning a defined object structure instead of a carefully nested array.

Reply:
Yeah, fancy one. But tbh I don't care much about it as it's used only for testing. I hate these mixed arrays too, well defined classes are much nicer, true. But here it comes to another drawback of PHP with the one class per PHP file model which I don't always like...

Reply:
I frequently ignore that convention when creating throw-away stuff for tests.  That's only a meaningful convention for autoloading, and since that doesn't have to be relevant for a one-off test class...

(Or, anonymous class, maybe?)  I agree for tests it's less of a concern, but still.